### PR TITLE
Feature/property move

### LINKED ISSUE #178: Reorder Properties in UI
Would be nice to be able to reorder properties, i.e. move them up or down. For example in the composite property. At the moment, the order is predetermined by the calls to `addProperty()`.

Reordering properties in a PropertyOwner is not sufficient, since deserialization is affected by the order of `addProperty()`
Suggestion: store some metadata to reorder properties upon deserialization.

### PR DESCRIPTION
**Pull request by petersteneteg**
Fixes #178

Changes proposed in this PR:
 * Added support for moving a property in the propertyowner

### CODE REVIEW

**Review by ResearchDaniel — DISMISSED** [review]

File: modules/postprocessing/src/processors/depthoffield.cpp [hunk 1]
```diff
@@ -212,8 +212,15 @@ void DepthOfField::process() {
             std::swap(nextOutImg_, prevOutImg_);
         }
         evalCount_++;
-        dispatchFront([this, camera, maxEvalCount, focusDepth]() {
-            moveCamera(camera, maxEvalCount, focusDepth);
+        dispatchFront([weakSelf = weak_from_this(), maxEvalCount, focusDepth]() {
```
Comment (martinfalk) [inline modules/postprocessing/src/processors/depthoffield.cpp:215]: this pattern should probably be mentioned in the docs

File: modules/pvm/src/pvmvolumereader.cpp [hunk 2]
```diff
@@ -99,9 +99,14 @@ std::shared_ptr<Volume> PVMVolumeReader::readPVMData(const std::filesystem::path
     unsigned char* parameter = nullptr;
     unsigned char* comment = nullptr;
 
-    unsigned char* pvmdata = readPVMvolume(filePath.string().c_str(), &udim.x, &udim.y, &udim.z,
-                                           &bytesPerVoxel, &spacing.x, &spacing.y, &spacing.z,
-                                           &description, &courtesy, &parameter, &comment);
+    auto file = filePath.string();  // convert to string before changing locale
+
+    auto locale = setlocale(LC_ALL, nullptr);  // save current locale
```
Comment (martinfalk) [inline modules/pvm/src/pvmvolumereader.cpp:104]: also applicable to Inviwo dat/raw files?

File: src/qt/editor/welcomewidget.cpp [hunk 3]
```diff
@@ -157,33 +182,62 @@ using Role = WorkspaceTreeModel::Role;
 
 namespace {
 
+bool contains(std::string_view str, std::string_view substr) {
+    auto it = std::search(
+        str.begin(), str.end(),
+        std::boyer_moore_horspool_searcher(
```
Comment (martinfalk) [inline src/qt/editor/welcomewidget.cpp:188]: Interesting ;-)

**Review by martinfalk — APPROVED** [review]

File: modules/postprocessing/src/processors/depthoffield.cpp (hunk 1 above)
Comment (petersteneteg) [reply modules/postprocessing/src/processors/depthoffield.cpp:215]: Well I would rather that know one did anything like this ;)

File: modules/pvm/src/pvmvolumereader.cpp (hunk 2 above)
Comment (petersteneteg) [reply modules/pvm/src/pvmvolumereader.cpp:104]: Has not had any problems with it for our files so far.

File: src/qt/editor/welcomewidget.cpp (hunk 3 above)
Comment (petersteneteg) [reply src/qt/editor/welcomewidget.cpp:188]: :) probably a premature optimization... but I might refactor it in the future

**Pull request merged**